app/models/katello/concerns/redhat_extensions.rb

```diff
@@ -44,16 +44,31 @@ def construct_name(family)
       end
 
       def medium_uri_with_content_uri(host, url = nil)
-        if host.try(:content_source) && (repo_details = kickstart_repo(host))
-          URI.parse(repo_details[:path])
+        if host.try(:content_source) && (repos = distribution_repositories(host)).present?
```

Comment:
Are the repos returned by `distribution_repositories` not already guaranteed to be valid distribution repos for the host and content view? Before, there didn't appear to be any validation checking.